x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -502,34 +514,34 @@ private void authorizeAction(
                         requestInfo,
                         authzInfo,
                         projectMetadata.getIndicesLookup(),
-                        ActionListener.wrap(
-                            authorizedIndices -> resolvedIndicesListener.onResponse(
-                                indicesAndAliasesResolver.resolve(action, request, projectMetadata, authorizedIndices)
-                            ),
-                            e -> {
-                                if (e instanceof InvalidIndexNameException
-                                    || e instanceof InvalidSelectorException
-                                    || e instanceof UnsupportedSelectorException) {
-                                    logger.debug(
-                                        () -> Strings.format(
-                                            "failed [%s] action authorization for [%s] due to [%s] exception",
-                                            action,
-                                            authentication,
-                                            e.getClass().getSimpleName()
+                        ActionListener.wrap(authorizedIndices -> {
+                            if (request instanceof IndicesRequest.Replaceable replaceable && replaceable.crossProjectResolvable()) {
+                                authorizedProjectsSupplier.getAuthorizedProjects(
+                                    ActionListener.wrap(
+                                        authorizedProjects -> resolvedIndicesListener.onResponse(
+                                            indicesAndAliasesResolver.resolve(
+                                                action,
+                                                request,
+                                                projectMetadata,
+                                                authorizedIndices,
+                                                authorizedProjects
+                                            )
                                         ),
-                                        e
-                                    );
-                                    listener.onFailure(e);
-                                    return;
-                                }
-                                auditTrail.accessDenied(requestId, authentication, action, request, authzInfo);
-                                if (e instanceof IndexNotFoundException) {
-                                    listener.onFailure(e);
-                                } else {
-                                    listener.onFailure(actionDenied(authentication, authzInfo, action, request, e));
-                                }
+                                        e -> authzResourceLoadFailure(requestId, requestInfo, authzInfo, auditTrail, e, listener)
```

Comment:
There might be a cleaner way to do this instead of having the authzResourceLoadFailure here and on the outer listener but I'm blanking on what listener util/magic to use for it.